### PR TITLE
MULE-18047: Add the component that generated a cursor provider (support/1.2.1)

### PR DESCRIPTION
**Pull request by szaffarano**
… (#674)

(cherry picked from commit 784a970a5a542ec6bb7dd77cdd4e3c2dc5daef44)
(cherry picked from commit c00415caf37b2151dbdb615f00a679182391fe8b)

### CODE REVIEW

File: api-changes.json [hunk 1]
```diff
@@ -2,7 +2,52 @@
...
+          "new": "field org.mule.runtime.api.util.MuleSystemProperties.ASYNC_TEST_CONNECTIVITY_TIMEOUT_PROPERTY",
+          "package": "org.mule.runtime.api.util",
+          "classSimpleName": "MuleSystemProperties",
+          "fieldName": "ASYNC_TEST_CONNECTIVITY_TIMEOUT_PROPERTY",
+          "elementKind": "field",
+          "justification": "Added a system property due to a bug"
+        },
+        {
+          "code": "java.method.added",
+          "new": "method org.mule.runtime.api.deployment.meta.MuleApplicationModel.MuleApplicationModelBuilder org.mule.runtime.api.deployment.meta.MuleApplicationModel.MuleApplicationModelBuilder::setDomain(java.lang.String)",
```
Comment (elrodro83) [inline api-changes.json:34]: this is not from this pr

File: api-changes.json [hunk 2]
```diff
@@ -2,7 +2,52 @@
...
+          "package": "org.mule.runtime.api.deployment.meta",
+          "classSimpleName": "MuleApplicationModelBuilder",
+          "methodName": "setDomain",
+          "elementKind": "method",
+          "justification": "Adding capability to set a domain name in the application model"
+        },
+        {
+          "code": "java.method.added",
+          "new": "method java.util.Optional<java.lang.String> org.mule.runtime.api.deployment.meta.MuleApplicationModel::getDomain()",
```
Comment (elrodro83) [inline api-changes.json:43]: this is not from this pr

File: api-changes.json (hunk 1 above)
Comment (szaffarano) [reply api-changes.json]: My bad, I had a problem while resolving a merge conflict.

File: api-changes.json (hunk 2 above)
Comment (szaffarano) [reply api-changes.json]: My bad, I had a problem while resolving a merge conflict.

**Pull request merged**